### ISSUE TITLE
Support table total functions e.g. SUBTOTAL(109,Table1[Value])

### ISSUE THREAD

**Issue by SabotageAndi** [ClosedXML/ClosedXML#271]
**Do you want to request a *feature* or report a *bug*?**
- [ ] Bug
- [x] Feature

**What is the current behavior?**
I have following formular in a cell: `=SUBTOTAL(109;[Value])`
If I access the value of the cell, I get this exception:
```
System.Exception: Identifier expected.
   at ClosedXML.Excel.CalcEngine.CalcEngine.GetToken() in C:\Projects\ClosedXML\ClosedXML\Excel\CalcEngine\CalcEngine.cs:line 589
   at ClosedXML.Excel.CalcEngine.CalcEngine.GetParameters() in C:\Projects\ClosedXML\ClosedXML\Excel\CalcEngine\CalcEngine.cs:line 640
   at ClosedXML.Excel.CalcEngine.CalcEngine.ParseAtom() in C:\Projects\ClosedXML\ClosedXML\Excel\CalcEngine\CalcEngine.cs:line 353
   at ClosedXML.Excel.CalcEngine.CalcEngine.ParseUnary() in C:\Projects\ClosedXML\ClosedXML\Excel\CalcEngine\CalcEngine.cs:line 332
   at ClosedXML.Excel.CalcEngine.CalcEngine.ParsePower() in C:\Projects\ClosedXML\ClosedXML\Excel\CalcEngine\CalcEngine.cs:line 313
   at ClosedXML.Excel.CalcEngine.CalcEngine.ParseMulDiv() in C:\Projects\ClosedXML\ClosedXML\Excel\CalcEngine\CalcEngine.cs:line 303
   at ClosedXML.Excel.CalcEngine.CalcEngine.ParseAddSub() in C:\Projects\ClosedXML\ClosedXML\Excel\CalcEngine\CalcEngine.cs:line 293
   at ClosedXML.Excel.CalcEngine.CalcEngine.ParseCompare() in C:\Projects\ClosedXML\ClosedXML\Excel\CalcEngine\CalcEngine.cs:line 283
   at ClosedXML.Excel.CalcEngine.CalcEngine.Parse(String expression) in C:\Projects\ClosedXML\ClosedXML\Excel\CalcEngine\CalcEngine.cs:line 81
   at ClosedXML.Excel.CalcEngine.ExpressionCache.get_Item(String expression) in C:\Projects\ClosedXML\ClosedXML\Excel\CalcEngine\ExpressionCache.cs:line 47
   at ClosedXML.Excel.CalcEngine.CalcEngine.Evaluate(String expression) in C:\Projects\ClosedXML\ClosedXML\Excel\CalcEngine\CalcEngine.cs:line 108
   at ClosedXML.Excel.XLCell.get_Value() in C:\Projects\ClosedXML\ClosedXML\Excel\Cells\XLCell.cs:line 407
   at ClosedXmlSubtotalRepro.Program.Main(String[] args) in G:\Work\ClosedXmlSubtotalRepro\ClosedXmlSubtotalRepro\Program.cs:line 16
```

This formular is used by Excel, if you format a range as table and check "Total Row" for it.

**If the current behavior is a bug, please provide the steps to reproduce and 
if possible a minimal demo of the problem with a sample spreadsheet.**

Repro Code to be executed against the provided Excel- File:
```
static void Main(string[] args)
{
	var xlWorkbook = new XLWorkbook("SubtotalRepro.xlsx");
	var sheet = xlWorkbook.Worksheets.First();
	var valueOfCell = sheet.Cell("A5").Value; //here is the exception thrown

	Console.WriteLine("Expected value: 6");
	Console.WriteLine($"Actual value: {valueOfCell}");
	Console.Read();
}
```

**What is the expected behavior or new feature?**
I expect to get the calculated value of the cell.

**Did this work in previous versions of our tool?  Which versions?**
I tried it with 0.87.1

- [x] I attached a sample spreadsheet.  (You can drag files on to this issue)
[SubtotalRepro.xlsx](https://github.com/ClosedXML/ClosedXML/files/916561/SubtotalRepro.xlsx)

**Comment by igitur** [comment]
Table column reference style (`[Value]`) is not supported yet.

**Comment by SabotageAndi** [comment]
Thanks! Good to know.
Is there any plan to add support for this in the future?

**Comment by igitur** [comment]
I don't really plan that far ahead.

**Comment by jahav** [comment]
The problematic formula in the provided worksheet `SUBTOTAL(109,Table1[Value])`. It needs structured reference that were implemented in #2251 (0.104) and also needs SUBTOTAL 109, that was implemented in #2473 (develop branch).

The provided repro now calculates correct value (forced recalculation, because workbook contain value calculated by Excel):
> Expected value: 6
> Actual value: 6
